### ISSUE TITLE
Filter for relations isnt working with permissions

### ISSUE THREAD

**Issue by Attacler** [directus/directus#15437]
### Describe the Bug

For a customer of mine we have a setup like this:
projects o2m ratings

When we want to get all projects that have a favorite true it returns:
![image](https://user-images.githubusercontent.com/42644889/188728585-92b30f40-9896-47b5-a43a-c81c0b69ce56.png)
The filter used for this:
filter: {"_and":[{"Competition":{"_eq":56},"ratings":{"favorite":{"_eq":true}}}]}

The 3th item does not contain any ratings because the permission rules prevent the user from reading it.
But then the question is, why is the latest one getting trough?

### To Reproduce

Create 1 collection named projects.
Create 1 collection named ratings.
Add a field to ratings with a boolean.
Create a o2m (projects -> ratings) one between them.
Create a role
Then create a rule with something like: 
{ "user_field": {"_eq": "$CURRENT_USER"}}
Allow read all for that role
After that create a user with a token.
Create some records in projects and create some ratings (1 based on the created user, the others based on the admin or someone else).

Try to filter on projects trough the api with the token like this:
{"ratings":{"favorite":{"_eq":true}}}

It will return all projects which have ratings with favorite to true regardless if the user can read the ratings itself.

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

Directus Standard Cloud

### What database are you using?

Directus Standard Cloud

### What browser are you using?

Chrome/Postman

### How are you deploying Directus?

Directus Standard Cloud

**Comment by licitdev** [comment]
### Notes
The read permissions to the nested collections should be appended into the db query.

**Comment by claytongulick** [comment]
Should this be considered a CVE?

Wouldn't it allow for speculative discovery?

**Comment by Attacler** [comment]
> Should this be considered a CVE?
> 
> Wouldn't it allow for speculative discovery?

Yes and no, i think it should be handled with a boolean or something if we want speculative discovery.
ATM it can mess up some stuff if you assume that filters work 100% with permissions.

**Comment by rijkvanzanten** [comment]
Linear: [ENG-195](https://linear.app/directus/issue/ENG-195/filter-for-relations-isnt-working-with-permissions)

**Comment by ComfortablyCoding** [comment]
I am no longer able to reproduce this based on the given information, as such I will proceed to close this for now. Happy to keep discussing in this thread, and we'll reopen if the issue can still be reproduced.

#### Reproduction Attempt

https://github.com/user-attachments/assets/d9cd164f-1d28-4306-bbff-6347ee1085a5
